doc/content/language.md

```diff
@@ -961,6 +961,31 @@ will print
 
 (note that the string is modified but not the fields `duration` and `bpm`).
 
+### Optional fields
+
+During the execution of your script, it can be useful to allow functions
+to receive records that may or may not have a specific field. This can
+be used, for instance, to model optional arguments.
+
+Liquidsoap supports that functionality using the `?.` syntax. Here's an example:
+
+```liquidsoap
+# This functions adds 1 to x unless options has a
+# add field in which case it adds this value
+def f(x, options) =
+  x + (options?.add ?? 1)
+end
+```
+
+The type of this function is:
+
+```liquidsoap
+f : (int, 'a.{add? : int?}) -> int = <fun>
```

Comment:
I don't think that the field is nullable here:
```suggestion
f : (int, 'a.{add? : int}) -> int = <fun>
```

Reply:
Nah that's actually the expected type, unfortunately. in `x?.foo ?? 123`, `x.foo` can either:
* Have no method
* Have a method that returns an `int`
* Have a method that returns `null`